controller/up.go

```diff
@@ -7,38 +7,100 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/railwayapp/cli/entity"
 	gitignore "github.com/railwayapp/cli/gateway"
 )
 
+var validIgnoreFile = map[string]bool{
+	".gitignore":     true,
+	".railwayignore": true,
+}
+
+var skipDirs = []string{
+	".git",
+	"node_modules",
+}
+
+type ignoreFile struct {
+	prefix string
+	ignore *gitignore.GitIgnore
+}
+
+func scanIgnoreFiles(src string) ([]ignoreFile, error) {
+	ret := []ignoreFile{}
```

Comment:
nit: ret should be more descriptive

Suggestion: `ignoredFilenames`

Reply:
Changed to `ignoreFiles`, wouldn't want to get an `ignoreFile` object confused w/ some list of file names.